applications: fix Matter Weather Station doc related with FactoryData (#13513)
nrfconnect/sdk-nrf: applications/matter_weather_station/README.rst

* ``release`` - Release version of the application.
  You can use this version to enable only the necessary application functionalities to optimize its performance.

.. note::
    Currently, this application supports only the ``release`` build type when `Building for the nRF7002 Wi-Fi expansion board`_.

.. note::
    `Selecting a build type`_ is optional.
    The ``debug`` build type is used by default if no build type is explicitly selected.

> Matter weather station build configuration overlays

Reviewer: I notice that for the weather station, we seem to have some sections in different places than we do for the samples. Could you move this new section and the build types sections to be under Configuration instead of Overview to match the samples.

Author: Sure.

===================================================

> * ``overlay-factory_data`` - factory data storage support enabled.

Reviewer: When it was a build type, it was based on the release version. Does the new overlay file work with both build types?

Author: Yes.

>   You can use this optional overlay to enable reading necessary factory data from a separate partition in the device non-volatile memory.

Reviewer 2: Maybe it is not fully related to this PR, but I noticed that we haven't mentioned that after enabling factory data support the BLE DFU between Weather station and other applications for Thingy53 is not possible anymore without reflashing via JLink. (Due to changes in the partition layout) Do you think it is worth adding here such info to this PR?

Reviewer 3: +1, I think it's an unwritten rule that all applications for Thingy should be compatible regarding the partitions layout, so it would be nice to point it out.

Author: Sure, I will add such a note.

>   You can use this optional overlay to enable reading necessary factory data from a separate partition in the device non-volatile memory.

Reviewer: ```suggestion
  You can use this optional configuration overlay to enable reading necessary factory data from a separate partition in the device non-volatile memory.
```

  You can use this version to enable only the necessary application functionalities to optimize its performance.

.. note::
    Currently, this application supports only the ``release`` build type when `Building for the nRF7002 Wi-Fi expansion board`_.

.. note::
    `Selecting a build type`_ is optional.
    The ``debug`` build type is used by default if no build type is explicitly selected.

.. _matter_weather_station_app_build_configuration_overlays:

Matter weather station build configuration overlays
===================================================

* ``overlay-factory_data`` - factory data storage support enabled.
  You can use this optional configuration overlay to enable reading necessary factory data from a separate partition in the device non-volatile memory.
  This way, you can read information such as product information, keys, and certificates, useful for example for Matter certification.
  See `Generating factory data`_ to learn how to put factory data into device's storage.
  To learn more about factory data, read the :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.

.. note::

>    Matter factory data support requires the dedicated partition layout, therefore if you build the application using ``overlay-factory_data``, it will not be compatible with other :ref:`Thingy:53 applications and samples <thingy53_compatible_applications>`.

Reviewer: ```suggestion
   Matter factory data support requires the dedicated partition layout.
   This means that if you build the application using the ``overlay-factory_data`` configuration overlay, it will not be compatible with other :ref:`Thingy:53 applications and samples <thingy53_compatible_applications>`.
```

Generating factory data
=======================

> To enable factory data support, you need to select the ``overlay-factory_data`` overlay from the available application :ref:`build configuration overlays <matter_weather_station_build_configuration_overlays>`.

Reviewer: ```suggestion
To enable factory data support, you need to select the ``overlay-factory_data`` configuration overlay from the available application :ref:`build configuration overlays <matter_weather_station_build_configuration_overlays>`.
```
You need to also add the target tag for the new heading. The missing tag is what is currently breaking the doc build.

Author: Right, thanks for the catch.
